scripts/typedoc/shopify-dev-renderer/extension-points.ts

```diff
@@ -88,3 +143,35 @@ export async function extensionPoints(paths: Paths, options: Options = {}) {
 
   console.log(`📄  Generated extension point docs to ${apiFile}.`);
 }
+
+async function buildHooksGraph(hooksIndex: string) {
+  const graph = await createDependencyGraph(hooksIndex);
+  const nodes: Node[] = [];
+
+  graph.forEach((value) => {
+    value.locals.forEach((value: any, key) => {
+      if (value.kind !== 'Imported') {
+        if (value.name == null) {
+          value.name = key;
+        }
+        nodes.push({value, module: undefined});
+      }
+    });
+  });
+  
+  const hooks = [
+    ...new Set(nodes.filter(({value}: any) => value.kind === 'Hook')),
+  ];
+
+  // Sort alphabetically (tsdoc seems to get this confused)
+  hooks.sort((aa: any, bb: any) => {
+    if (aa.value.name > bb.value.name) {
+      return 1;
+    } else if (aa.value.name < bb.value.name) {
+      return -1;
+    } else {
+      return 0;
+    }
```

Comment:
Might be a cool place to use [String.prototype.localeCompare()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare)

```suggestion
    return aa.value.name.localeCompare(bb.value.name)
```